service.go

```diff
@@ -92,10 +100,11 @@ func (s *Service) dispatchRequest(ctx context.Context, conn net.Conn) {
 	for {
 		m, err := mongowire.ReadMessage(conn)
 		if err != nil {
-			if err == io.EOF {
+			if errors.Cause(err) == io.EOF {
 				return
 			}
-			grip.Warning("problem reading message")
+			grip.Error(errors.Wrap(err, "problem reading message"))
```

Comment:
I think we should probably use, in general message.WrapError(err, <msg>) rather than errors wrap here, because it's more clear, maybe?)